core/src/plugins/access.ftp/class.ftpAccessDriver.php

```diff
@@ -214,6 +214,20 @@ public function uploadActions($action, $httpVars, $filesVars)
 
     }
 
+    // Checks if a file belongs to currently logged in FTP user
+    private function isFileOwner($path)
+    {
+        $ftp = new ftpAccessWrapper();
+        $stat = $ftp->url_stat($path, 2);
+        $urlParts = AJXP_Utils::safeParseUrl($path);
+        $repository = ConfService::getRepositoryById($urlParts["host"]);
+        $credentials = AJXP_Safe::tryLoadingCredentialsFromSources($urlParts, $repository);
+        if (empty($credentials["user"]))
+            return is_writable($path);
```

Comment:
Here we don't use the $stat value, so we should make sure to have this case return earlier in the function (before making an unecessary call to url_stat).